code/datums/components/anti_magic.dm

```diff
@@ -1,26 +1,43 @@
 /datum/component/anti_magic
 	var/magic = FALSE
 	var/holy = FALSE
+	var/charges = INFINITY
+	var/blocks_self = TRUE
+	var/datum/callback/reaction
+	var/datum/callback/expire
 
-/datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE)
+/datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE, _charges = -1, _blocks_self = TRUE, datum/callback/_reaction, datum/callback/_expire)
```

Comment:
You didn't modify charges here, honestly you only need the type definition

Reply:
Will fix